pkg/query/queryexecute/execute.go

```diff
@@ -18,31 +18,32 @@ import (
 	"github.com/turbot/steampipe/pkg/utils"
 )
 
-func RunInteractiveSession(ctx context.Context, initData *query.InitData) {
+func RunInteractiveSession(ctx context.Context, initData *query.InitData) error {
 	utils.LogTime("execute.RunInteractiveSession start")
 	defer utils.LogTime("execute.RunInteractiveSession end")
 
 	// the db executor sends result data over resultsStreamer
-	resultsStreamer, err := interactive.RunInteractivePrompt(ctx, initData)
+	result, err := interactive.RunInteractivePrompt(ctx, initData)
```

Comment:
no keen on this - result has an error, but we also return an error, and we handle the 2 errors differently - this is a bit confusing.

also, if `err` is non nil we just fail - what is the exit code in this case?

Reply:
fixed